Framework/Algorithms/test/CalcCountRateTest.h

```diff
@@ -0,0 +1,478 @@
+
+    CalcCountRateTester alg;
+    alg.initialize();
+
+    //-------- check defaults
+    alg.setOutLogParameters(sws);
+
+    alg.getAlgLogSettings(numLogSteps, pNormLog);
+    TS_ASSERT_EQUALS(numLogSteps, 200);
+    TS_ASSERT(!pNormLog);
+
+    //-------- check numLogSteps
+    alg.setProperty("NumTimeSteps", 100);
+
+    alg.setOutLogParameters(sws);
+
+    alg.getAlgLogSettings(numLogSteps, pNormLog);
+    TS_ASSERT_EQUALS(numLogSteps, 100);
+    TS_ASSERT(!pNormLog);
+
+    //-------- check numLogSteps, normalization log ignored
+    alg.setProperty("NumTimeSteps", 120);
+    alg.setProperty("NormalizeTheRate", true);
+    alg.setProperty("UseLogDerivative", false);
+    alg.setProperty("UseNormLogGranularity", true);
+
+    alg.setOutLogParameters(sws);
+    alg.getAlgLogSettings(numLogSteps, pNormLog);
+    TS_ASSERT_EQUALS(numLogSteps, 120);
+    TS_ASSERT(!pNormLog);
+    TS_ASSERT(!alg.notmalizeCountRate());
+
+    // Check time series log outside of the data range
+    auto pTime_log = new Kernel::TimeSeriesProperty<double>("proton_charge");
+    Kernel::DateAndTime first("2015-11-30T16:17:10");
+    std::vector<Kernel::DateAndTime> times(140);
+    std::vector<double> values(140);
+
+    for (size_t i = 0; i < 140; ++i) {
+      times[i] = first + double(i);
+      values[i] = double(i);
+    }
+
+    // DateAndTime("2010-01-01T00:00:00")
+
+    pTime_log->addValues(times, values);
+    sws->mutableRun().addProperty(pTime_log, true);
+
+    alg.setOutLogParameters(sws);
+    alg.getAlgLogSettings(numLogSteps, pNormLog);
+    TS_ASSERT_EQUALS(numLogSteps, 120);
+    TS_ASSERT(!pNormLog);
+    TS_ASSERT(!alg.notmalizeCountRate());
+    TS_ASSERT(!alg.useLogDerivative());
+
+    // Check correct date and time
+    first = Kernel::DateAndTime("2010-01-01T00:00:00");
+    times.resize(240);
+    values.resize(240);
+    for (size_t i = 0; i < 240; ++i) {
+      times[i] = first - 20. + double(i);
+      values[i] = double(i);
+    }
+    pTime_log->replaceValues(times, values);
+
+    alg.setOutLogParameters(sws);
+    alg.getAlgLogSettings(numLogSteps, pNormLog);
+    TS_ASSERT_EQUALS(numLogSteps, 99);
+    TS_ASSERT(pNormLog);
+    TS_ASSERT(alg.notmalizeCountRate());
+    TS_ASSERT(!alg.useLogDerivative());
+
+    // check useLogDerivative
+    alg.setProperty("UseLogDerivative", true);
+    //
+    alg.setOutLogParameters(sws);
+    alg.getAlgLogSettings(numLogSteps, pNormLog);
+    TS_ASSERT_EQUALS(numLogSteps, 100);
+    TS_ASSERT(pNormLog);
+    TS_ASSERT(alg.notmalizeCountRate());
+    TS_ASSERT(alg.useLogDerivative());
+  }
+
+  void test_processing() {
+
+    auto sws = build_test_ws(true);
+
+    CalcCountRateTester alg;
+    alg.initialize();
+
+    alg.setProperty("NumTimeSteps", 120);
+    alg.setProperty("NormalizeTheRate", true);
+    alg.setProperty("UseLogDerivative", true);
+    alg.setProperty("UseNormLogGranularity", true);
+
+    alg.setProperty("Workspace", sws);
+
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
```

Comment:
To my understanding, calling `execute()` almost never throws (even if the algorithm actually does!) unless you call `alg.setRethrows(true)` first. So I recommend doing that in all the affected tests.